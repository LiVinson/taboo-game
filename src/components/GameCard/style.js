import styled from "styled-components"

export const InstructionsText = styled.p`
    font-size:2rem;
    color: ${props => props.theme.color.grayDark2};
    text-align:left;
    &:not(:last-child){
        margin-bottom:1rem;
    }
`

export const KeyWord = styled.span`
    color: ${props => props.theme.color.accent2};

`