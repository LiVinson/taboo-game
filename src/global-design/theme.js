const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
