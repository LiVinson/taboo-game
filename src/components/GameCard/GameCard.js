import React from 'react'
import PropType from 'prop-types'
import { ButtonTabooCard, TabooCard } from 'components/shared/TabooCard'
import { InstructionsText, KeyWord } from './style'

//current card object, skip and next callback.
export const GiverGameCard = ({ currentCard, changeCardStatus, isPending }) => {
	const buttonInfo = [
		{
			text: 'Skip!',
			disabled: isPending,
			onClick: () => {
				console.log('skip card')
				changeCardStatus('skipped')
			},
		},
		{
			text: 'Next!',
			disabled: isPending,

			onClick: () => {
				console.log("next")
				changeCardStatus('correct')
			},
		},
	]


	return <ButtonTabooCard buttons={buttonInfo} tabooWord={currentCard.word} list={currentCard.tabooList} />
}

export const WatcherGameCard = () => {
	const buttonInfo = [
		{
			text: 'Buzzer!',
			onClick: () => {
				console.log('Buzz')
			},
		},
	]

	const card = {
		tabooWord: 'Simba',
		words: ['Lion King', 'Disney', 'Mufasa', 'Pride Rock', 'Nala'],
	}

	return <ButtonTabooCard buttons={buttonInfo} tabooWord={card.tabooWord} list={card.words} />
}

export const TeamGameCard = ({ role, giver, watcher }) => {
	return (
		<React.Fragment>
			{role === 'giverTeam' ? (
				<TabooCard tabooWord="Guess!">
					<InstructionsText>
						It's your teams turn to guess the word! <KeyWord>{giver.name}</KeyWord> is giving the clues and{' '}
						<KeyWord>{watcher.name}</KeyWord> will be making sure they don't say anything Taboo!
					</InstructionsText>
				</TabooCard>
			) : (
				<TabooCard tabooWord="Relax!">
					<InstructionsText>
						It’s the other team’s turn to give clues and guess! <KeyWord>{watcher.name}</KeyWord> will be
						watching to make sure <KeyWord>{giver.name}</KeyWord> doesn’t say any Taboo words.
					</InstructionsText>
				</TabooCard>
			)}
		</React.Fragment>
	)
}

TeamGameCard.propType = {
	role: PropType.string.isRequired,
	giver: PropType.string.isRequired,
	watcher: PropType.string.isRequired,
}
