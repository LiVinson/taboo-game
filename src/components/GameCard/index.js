import { GameCard } from "./GameCard"

export { GameCard }