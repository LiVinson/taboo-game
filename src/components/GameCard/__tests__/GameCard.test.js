import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../../../global-design/theme'
import { GiverGameCard, WatcherGameCard, TeamGameCard } from '../GameCard'

describe('GameCard functionality and rendering', () => {
	const defaultProps = {
		giver: {},
		watcher: {},
		role: '',
		currentCard: {
			word: 'Will Smith',
			tabooList: ['Fresh Prince', 'Philadelphia', 'Jazzy Jeff', 'Independence Day', 'Jada Pinkett'],
		},
		isPending: false,
		changeCardStatus: jest.fn(),
		cardsRemainingMsg: '',
		error: null,
	}

	test('GiverGameCard renders correctly', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<GiverGameCard {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})

	test("Selecting 'Skip' calls props.changeCardStatus with correct status", () => {
		const props = {
			...defaultProps,
		}

		const wrapper = mount(
			<ThemeProvider theme={theme}>
				<GiverGameCard {...props} />
			</ThemeProvider>
		)

		wrapper.find('button').at(0).simulate('click')
		expect(props.changeCardStatus).toHaveBeenCalledWith('skipped')
	})

	test("Selecting 'Next' calls props.changeCardStatus with correct status", () => {
		const props = {
			...defaultProps,
		}

		const wrapper = mount(
			<ThemeProvider theme={theme}>
				<GiverGameCard {...props} />
			</ThemeProvider>
		)

		wrapper.find('button').at(1).simulate('click')
		expect(props.changeCardStatus).toHaveBeenCalledWith('correct')
	})

	test('WatcherGameCard renders correctly', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<WatcherGameCard {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})

	test("Selecting 'Buzzer' calls props.changeCardStatus with correct status", () => {
		const props = {
			...defaultProps,
		}

		const wrapper = mount(
			<ThemeProvider theme={theme}>
				<WatcherGameCard {...props} />
			</ThemeProvider>
		)

		wrapper.find('button').at(0).simulate('click')
		expect(props.changeCardStatus).toHaveBeenCalledWith('discard')
	})

	test('TeamGameCard renders correctly for watcherTeam', () => {
		const props = {
			role: 'watcherTeam',
			giver: {
				name: 'Joe',
			},
			watcher: {
				name: 'Sam',
			},
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<TeamGameCard {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})

	test('TeamGameCard renders correctly for giverTeam', () => {
		const props = {
			role: 'giverTeam',
			giver: {
				name: 'Joe',
			},
			watcher: {
				name: 'Sam',
			},
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<TeamGameCard {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})
})
