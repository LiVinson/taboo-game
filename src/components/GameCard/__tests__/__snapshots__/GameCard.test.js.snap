// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCard functionality and rendering GiverGameCard renders correctly 1`] = `
<div
  className="sc-AxjAm hDlFTS"
>
  <div
    className="sc-AxiKw dGnLMo"
  >
    Will Smith
  </div>
  <div
    className="sc-AxhCb JZaCe"
  >
    <ul
      className="sc-Axmtr dHjpTi"
    >
      <li
        className="sc-AxhUy kWmTzm"
      >
        Fresh Prince
      </li>
      <li
        className="sc-AxhUy kWmTzm"
      >
        Philadelphia
      </li>
      <li
        className="sc-AxhUy kWmTzm"
      >
        Jazzy Jeff
      </li>
      <li
        className="sc-AxhUy kWmTzm"
      >
        Independence Day
      </li>
      <li
        className="sc-AxhUy kWmTzm"
      >
        Jada Pinkett
      </li>
    </ul>
  </div>
  <div
    className="sc-fzplWN buDbqD"
  >
    <button
      className="sc-AxmLO gmmVt"
      disabled={false}
      onClick={[Function]}
    >
      Skip!
    </button>
    <button
      className="sc-AxmLO sc-fzozJi jlLOkV"
      disabled={false}
      onClick={[Function]}
    >
      Next!
    </button>
  </div>
</div>
`;

exports[`GameCard functionality and rendering TeamGameCard renders correctly for giverTeam 1`] = `
<div
  className="sc-AxjAm hDlFTS"
>
  <div
    className="sc-AxiKw dGnLMo"
  >
    Guess!
  </div>
  <div
    className="sc-AxhCb JZaCe"
  >
    <p
      className="sc-fzokOt kxxFRs"
    >
      It's your teams turn to guess the word! 
      <span
        className="sc-fzqBZW hgCjBU"
      >
        Joe
      </span>
       is giving the clues and
       
      <span
        className="sc-fzqBZW hgCjBU"
      >
        Sam
      </span>
       will be making sure they don't say anything Taboo!
    </p>
  </div>
</div>
`;

exports[`GameCard functionality and rendering TeamGameCard renders correctly for watcherTeam 1`] = `
<div
  className="sc-AxjAm hDlFTS"
>
  <div
    className="sc-AxiKw dGnLMo"
  >
    Relax!
  </div>
  <div
    className="sc-AxhCb JZaCe"
  >
    <p
      className="sc-fzokOt kxxFRs"
    >
      It’s the other team’s turn to give clues and guess, so don't say anything!
       
      <span
        className="sc-fzqBZW hgCjBU"
      >
        Sam
      </span>
       will be watching to make sure 
      <span
        className="sc-fzqBZW hgCjBU"
      >
        Joe
      </span>
       
      doesn’t say any Taboo words.
    </p>
  </div>
</div>
`;

exports[`GameCard functionality and rendering WatcherGameCard renders correctly 1`] = `
<div
  className="sc-AxjAm hDlFTS"
>
  <div
    className="sc-AxiKw dGnLMo"
  >
    Will Smith
  </div>
  <div
    className="sc-AxhCb JZaCe"
  >
    <ul
      className="sc-Axmtr dHjpTi"
    >
      <li
        className="sc-AxhUy kWmTzm"
      >
        Fresh Prince
      </li>
      <li
        className="sc-AxhUy kWmTzm"
      >
        Philadelphia
      </li>
      <li
        className="sc-AxhUy kWmTzm"
      >
        Jazzy Jeff
      </li>
      <li
        className="sc-AxhUy kWmTzm"
      >
        Independence Day
      </li>
      <li
        className="sc-AxhUy kWmTzm"
      >
        Jada Pinkett
      </li>
    </ul>
  </div>
  <div
    className="sc-fzplWN buDbqD"
  >
    <button
      className="sc-AxmLO sc-fzoLsD cyAtQH"
      disabled={false}
      onClick={[Function]}
    >
      Buzzer!
    </button>
  </div>
</div>
`;
